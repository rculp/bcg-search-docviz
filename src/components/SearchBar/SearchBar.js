import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Input from 'components/Input/Input';
import Button from 'components/Button/Button';

import './SearchBar.scss';

class SearchBar extends Component {
  handleChange = (event) => {
    const { changeHandler } = this.props;
    changeHandler(event.target.value);
  };

  fetchResults = () => {
    const { apiCall, searchValue } = this.props;
    apiCall(searchValue);
  };

  render = () => {
    const { searchValue, isLoading, isDisabled } = this.props;
    return (
      <Input
        size="massive"
        placeholder="Search..."
        id="query"
        loading={isLoading}
        disabled={isDisabled}
        value={searchValue}
        onChange={this.handleChange}
        icon="search"
        iconPosition="left"
        label={<Button type="submit" onClick={this.fetchResults}>Submit</Button>}
        labelPosition="right"
      />
    );
  };
}

SearchBar.propTypes = {
  searchValue: PropTypes.string,
  isLoading: PropTypes.bool,
  isDisabled: PropTypes.bool,
  changeHandler: PropTypes.func.isRequired,
  apiCall: PropTypes.func.isRequired
};

SearchBar.defaultProps = {
  searchValue: '',
  isLoading: false,
  isDisabled: false
};

export default SearchBar;
