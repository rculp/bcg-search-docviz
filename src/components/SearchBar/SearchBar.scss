@import 'src/global-styles/colors';
@import 'src/global-styles/responsive';

@keyframes fadein {
  0% { opacity: 0.7; }
  66% { opacity: 1; }
  100% { opacity: 0.7; }
}

@keyframes fadeout {
  0% { opacity: 1; }
  66% { opacity: 0.7; }
  100% { opacity: 1; }
}


.searchBar {
  // keep
}

.ui.button.searchButton {
  @media (max-width: map-get($grid-breakpoints, sm)) {
    display: none;
  }
}

.iconContainer {
  &.searchIcon {
    padding: 0 8px;
  }

  &.loadIcons {
    padding: 4px 0;
  }

  .icon.arrow {
    font-size: 24px;
    vertical-align: middle;
    margin: 0;
    width: 1em;
  }

  
  .icon.circle {
    font-size: 8px;
    vertical-align: middle;
    margin-right: 8px;
    width: initial;
    
    &.right {
      margin-right: 0;
    } 

    &.left,
    &.right {
      animation: 1s ease 0s normal forwards 1 fadein;
      animation-iteration-count: infinite;
    }

    &.middle {
      animation: 1s ease 0s normal forwards 1 fadeout;
      animation-iteration-count: infinite;
    }
  }
}

.ui.disabled.input {
  opacity: 1;
}

.ui.label.button {
  background: $bcg-green;
  color: $white;
  padding: 16px 8px !important;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;

  &:focus {
    outline-color: -webkit-focus-ring-color;
    outline-style: auto;
  }

  &:hover,
  &:focus {
    color: $white;
    background: darken($bcg-green, 8%);
    
    .icon {
      opacity: 1;
    }
  }
}

.ui.form .ui.input {
  color: $gray--dark;
  font-size: 16px;

  &:focus {
    color: $gray--darker;
  }

  input {
    background: $white;
    border: 1px solid $gray--light;
    color: $gray--dark;
    font-family: HendersonBCGSans-Regular;
    font-size: 16px;
    letter-spacing: 0;
    line-height: 24px;
    
    @media (max-width: map-get($grid-breakpoints, sm)) {
      border-right-color: $gray--light !important;
      border-bottom-right-radius: 0.285714rem !important;
      border-top-right-radius: 0.285714rem !important;
      font-size: 14px;
      line-height: 21px;
    }

    &:hover {
      border-color: darken($gray, 8%);
      @media (max-width: map-get($grid-breakpoints, sm)) {
        border-right-color: darken($gray, 8%) !important;
      }
    }

    &:focus {
      border-color: $bcg-green;
      color: $gray--darker;
      outline-color: -webkit-focus-ring-color;
      outline-style: auto;
      @media (max-width: map-get($grid-breakpoints, sm)) {
        border-right-color: $bcg-green !important;
      }
    }
  }
}