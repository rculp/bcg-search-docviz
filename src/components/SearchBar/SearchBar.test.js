import { shallow } from 'enzyme';

import React from 'react';

import SearchBar from './SearchBar';

describe('SearchBar', () => {
  it('renders and matches our snapshot', () => {
    const component = shallow(
      <SearchBar />
    );
    expect(component).toMatchSnapshot();
  });
});
