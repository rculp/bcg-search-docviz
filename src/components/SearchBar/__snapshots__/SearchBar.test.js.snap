// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders and matches our snapshot 1`] = `
<Input
  disabled={false}
  icon="search"
  iconPosition="left"
  id="query"
  label={
    <Button
      as="button"
      className="searchButton"
      icon={true}
      onClick={[Function]}
      role="button"
      type="submit"
    >
      <div
        className="iconContainer searchIcon"
      >
        <Icon
          as="i"
          name="arrow right"
        />
      </div>
    </Button>
  }
  labelPosition="right"
  onChange={[Function]}
  placeholder="Try to be as descriptive as possible"
  size="massive"
  type="text"
  value=""
/>
`;
